dgbpy/torch_classes.py

```diff
@@ -746,82 +746,129 @@ def transform(self, arr):
             return np.rot90(arr,self.aug_count,self.aug_dims).copy()
 
 class RandomGaussianNoise():
-    def __init__(self, p=0.3, std=.3):
+    def __init__(self, p=0.3, std=0.1):
         self.p = p
         self.std = std
 
-    def update_pars(self, transform_dict):
-        self.p = float(transform_dict['p'])
-        self.std = float(transform_dict['std'])
+    def transformLabel(self, info):
+        return False
         
-    def __call__(self, image, label, image_only=True):
-        self.noise = np.random.normal(loc = 0, scale = self.std, size = image.shape).astype('float32')
+    def __call__(self, image=None, label=None):
         if self.p > np.random.uniform(0,1):
+            self.noise = np.random.normal(loc = 0, scale = self.std, size = image.shape).astype('float32')
             return self.transform(image), label
-        else:
-            return image, label
+        return image, label
 
     def transform(self, arr):
         arr = self.noise + arr
         return arr
 
+class ScaleTransform():
+    def __init__(self):
+        from dgbpy.dgbscikit import scale
+        self.scale = scale
+
+    def transformLabel(self, info):
+        return dgbhdf5.doOutputScaling(info)       
+
+    def __call__(self, image=None, label=None):
+        if isinstance(label, type(None)):
+            return self.transform(image), label
+        return self.transform(image), self.transform(label)
+        
+class Normalization(ScaleTransform):
+    def __init__(self):
+        super().__init__()
+        from dgbpy.dgbscikit import getNewMinMaxScaler
+        self.getNewMinMaxScaler = getNewMinMaxScaler
+
+    def transform(self, arr):
+        self.scaler = self.getNewMinMaxScaler(arr)
+        return self.scale(arr, self.scaler)
+
+class StandardScaler(ScaleTransform):
+    def __init__(self):
+        super().__init__()
+        from dgbpy.dgbscikit import getScaler
+        self.getScaler = getScaler
+
+    def transform(self, arr):
+        self.scaler = self.getScaler(arr, False)
+        return self.scale(arr, self.scaler)
+
+class MinMaxScaler(ScaleTransform):
+    def __init__(self):
+        super().__init__()
+        from dgbpy.dgbscikit import getNewMinMaxScaler
+        self.getNewMinMaxScaler = getNewMinMaxScaler
+ 
+    def transform(self, arr):
+        self.scaler = self.getNewMinMaxScaler(arr, maxout=255)
+        return self.scale(arr, self.scaler)
 
 all_transforms = {
     'RandomFlip': RandomFlip,
     'RandomGaussianNoise': RandomGaussianNoise,
+    dgbkeys.localstdtypestr: StandardScaler,
+    dgbkeys.normalizetypestr: Normalization,
+    dgbkeys.minmaxtypestr: MinMaxScaler
 }
 
 class TransformComposefromList():
-    def __init__(self, transforms):
+    """
+        Applies all the transform from a list to the data.
+    """
+    def __init__(self, transforms, info):
         if not hasattr(transforms, '__iter__'):
```

Comment:
if not isinstance(transforms, list):

Reply:
Thanks a lot...
I was actually making it flexible incase users insert a tuple of transforms as well.
I will keep it to just list and tuple.
Using a dictionary with this will also give issues..